pkg/controllers/overrider/clusterresource_controller_integration_test.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package overrider
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+)
+
+var _ = Describe("Test ClusterResourceOverride controller logic", func() {
+	var cro *fleetv1alpha1.ClusterResourceOverride
+	var testCROName string
+
+	BeforeEach(func() {
+		testCROName = fmt.Sprintf("test-cro-%s", utils.RandStr())
+		cro = getClusterResourceOverride(testCROName)
+	})
+
+	AfterEach(func() {
+		By("Deleting the CRO")
+		Expect(k8sClient.Delete(ctx, cro)).Should(SatisfyAny(Succeed(), &utils.NotFoundMatcher{}))
+	})
+
+	It("Test create a new CRO should result in one new snapshot", func() {
+		By("Creating a new CRO")
+		Expect(k8sClient.Create(ctx, cro)).Should(Succeed())
+		By("Checking if the finalizer is added to the CRO")
+		Eventually(func() error {
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: cro.Name}, cro)
+			if err != nil {
+				return err
+			}
+			if !controllerutil.ContainsFinalizer(cro, fleetv1alpha1.OverrideFinalizer) {
+				return fmt.Errorf("finalizer not added")
+			}
+			return nil
+		}, eventuallyTimeout, interval).Should(Succeed(), "snapshot should have finalizer")
+
+		By("Checking if a new snapshot is created")
+		snapshot := getClusterResourceOverrideSnapshot(testCROName, 1) //index start from 1
+		Eventually(func() error {
+			return k8sClient.Get(ctx, types.NamespacedName{Name: snapshot.Name}, snapshot)
+		}, eventuallyTimeout, interval).Should(Succeed(), "snapshot should exist")
+		By("Checking if the label is correct")
+		diff := cmp.Diff(map[string]string{
+			fleetv1alpha1.OverrideIndexLabel:    strconv.Itoa(1),
+			fleetv1beta1.IsLatestSnapshotLabel:  "true",
+			fleetv1alpha1.OverrideTrackingLabel: testCROName,
+		}, snapshot.GetLabels())
+		Expect(diff).Should(BeEmpty(), diff)
```

Comment:
```suggestion
		Expect(diff).Should(BeEmpty(), diff, "Snapshot label mismatch (-want, +got)")
```
need to include error message

Reply:
isn't diff already the error message?

Reply:
yes, but the diff don't provide enough information, eg, which two objects? and what's the "-" and "+". it will be hard to debug the failure unless we read the code.

btw, If i remember it correctly, the expect will print the "diff" automatically.